### DISCUSSION TITLE
Power up printer via octofarm

### DISCUSSION THREAD

**Discussion by taker218** [OctoFarm/OctoFarm#993] [Q&A]
Hello,

I'm using OctoFarm for a while now, but it seems that with one of the latest releases, it's not possible to power up a wifi plug via the interface.

I can't seem to find the settings anymore nor a button.

I have all my printers connected to wifi plugs running tasmota so I can turn on/shut down via octoprint and now I always have to go to the webinterface of the printer I want to use to turn it on rather then doing that via octofarm like I used to do.

Am I missing something or was it thrown out?

Best regards

**Comment by NotExpectedYet** [comment]
In Printer Manager -> Under the Manager dropdown there's the Settngs Modal. The power settings are there as before. 

![image](https://user-images.githubusercontent.com/61137988/170174576-7c8c30e1-f25b-4193-a544-7ba47ecc22c4.png)

![image](https://user-images.githubusercontent.com/61137988/170174682-6d5184cd-41b6-4092-85c4-238dbfc6d8cf.png)

All I have thrown out is the toggle. 

![image](https://user-images.githubusercontent.com/61137988/170174727-87b9a6d6-d1a0-4c43-b3b3-ab2772e58fa8.png)

Once those are setup the dropdown will appear under the Menu after a page refresh of the browser for the printer. 

You can either turn on or off there, and you should also see a little power status indicator next to the name. 

![image](https://user-images.githubusercontent.com/61137988/170174780-7b593591-784b-4fee-9298-6619f31ecd6e.png)


There's a few little changes to how this all works now regarding other buttons is why I never put in the toggle. 

When you press quick connect, it will first try to power on your printer if able. When you press print from anywhere in the interface, inc. fle manager and the bulk print options it will try to power on then try to connect your printer. 

Hope that helps

**Reply by taker218** [reply]
Ah I see!
Haven't seen it there, but I have found it.

Only problem when I want to save it, it gives me an error:

```
TypeError ERROR (undefined):
Please report this error to [OctoFarm Issues](https://github.com/octofarm/octofarm/issues)!
t.status is undefined
FILE INFO
LINE: 20
COL: 39734
FILE:
http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js
STACK:
createServerResponseMessageForSaveAction@http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js:20:39734 setupSaveButton/<@http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js:20:38911
```

**Reply by NotExpectedYet** [reply]
> Ah I see! Haven't seen it there, but I have found it.
> 
> Only problem when I want to save it, it gives me an error:
> 
> ```
> TypeError ERROR (undefined):
> Please report this error to [OctoFarm Issues](https://github.com/octofarm/octofarm/issues)!
> t.status is undefined
> FILE INFO
> LINE: 20
> COL: 39734
> FILE:
> http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js
> STACK:
> createServerResponseMessageForSaveAction@http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js:20:39734 setupSaveButton/<@http://192.168.178.7:4000/assets/dist/printer-management.runner.min.js:20:38911
> ```

That's client side by the looks, I'll open an issue to get it fixed in 1.2.3 this weekend. It "should" have saved the server side and still activate client side after refreshing the page. 

Issue #995 

Sorry about that! Seems I missed quite a lot this 1.2 release xD

**Reply by NotExpectedYet** [reply]
If it hasn't activated then would you mind adding a recent log dump to the issue for me please?

**Reply by taker218** [reply]
Sure thing, it seems like it didn't save the settings, just confirmed it from another device and the settings I put in are not there.

**Reply by NotExpectedYet** [reply]
> Sure thing, it seems like it didn't save the settings, just confirmed it from another device and the settings I put in are not there.

Will sort it cheers! Dunno what I broke there, as I've got it setup on my own instance at home on V1.2.
